LEAF_Request_Portal/sources/System.php

```diff
@@ -289,413 +264,153 @@ public function updateGroup(int $groupID): array
+                    FROM `users`
+                    WHERE `backupID` <> ""
+                    AND `groupID` = :groupID';
+
+            $this->db->prepared_query($sql, $vars);
+
+            foreach ($resEmp as $emp) {
+                if ($emp['userName'] != '') {
+                    $vars = array(':userID' => $emp['userName'],
+                        ':groupID' => $groupID,);
+
+                    $this->db->prepared_query('INSERT INTO users (userID, groupID, backupID)
+                                                        VALUES (:userID, :groupID, "")
+                                                        ON DUPLICATE KEY UPDATE userID=:userID, groupID=:groupID', $vars);
+
+                    // include the backups of employees
+                    $res = $this->db->prepared_query('SELECT * FROM users WHERE userID=:userID AND groupID=:groupID', $vars);
+                    if ($res[0]['active'] == 1) {
+                        $backups = $employee->getBackups($emp['empUID']);
+                        foreach ($backups as $backup) {
+                            $vars = array(':userID' => $backup['userName'],
+                                ':groupID' => $groupID,
+                                ':backupID' => $emp['userName'],);
+
+                            // Add backupID check for updates
+                            $this->db->prepared_query('INSERT INTO users (userID, groupID, backupID)
+                                                        VALUES (:userID, :groupID, :backupID)
+                                                        ON DUPLICATE KEY UPDATE userID=:userID, groupID=:groupID', $vars);
+                        }
+                    }
+                }
+            }
+            $return_value = "groupID: {$groupID} imported";
         }
 
         return $return_value;
     }
 
-    /**
-     * @param int $groupID
-     *
-     * @return array
-     *
-     * Created at: 6/30/2023, 1:25:25 PM (America/New_York)
-     */
-    private function deleteCatPrivs(int $groupID): array
+    public function getServices()
     {
-        $vars = array(':groupID' => $groupID);
-        $sql = 'DELETE
-                FROM `category_privs`
-                WHERE `groupID` = :groupID';
-
-        $return_value = $this->db->pdo_delete_query($sql, $vars);
-
-        return $return_value;
+        return $this->db->prepared_query('SELECT groupID as parentID,
+        							serviceID as groupID,
+        							service as groupTitle,
+        							abbreviatedService as groupAbbreviation
+        							FROM services
+        							ORDER BY groupTitle ASC', array());
     }
 
     /**
-     * @param int $groupID
-     *
-     * @return array
+     * Get the current database version
      *
-     * Created at: 6/30/2023, 1:25:38 PM (America/New_York)
+     * @return string the current database version
      */
-    private function getCatPrivs(int $groupID): array
+    public function getDatabaseVersion()
```

Comment:
Return value and same question about changing the name of the function from above.

Reply:
I didn't change the name I rearranged. So git doesn't like that and thinks it changed.